lw-oci-integration/tf/README.md

```diff
@@ -1,34 +1,103 @@
-```
-region="<YOUR HOME REGION  (e.g. us-ashburn-1)>"
-tenancy_ocid="<YOUR OCI TEANCY OCID>"
-user_ocid="<YOUR OCI USER OCID>"
-fingerprint="<YOUR KEY FINGER PRINT>"
-private_key_path="<YOUR KEY KEY PATH (e.g. ~/.oci/myocikey.pem)>"
-
-```
-3. Once the file is updated, run the following command to initialize Terraform:
-`terraform init`
-4. Now test the Terraform using this command:
-```
-terraform plan
-```
-5. If everything looks good, apply the change using the following command:
-```
-terraform apply -auto-approve
-```
-6. After the plan is applied, a Lacework cloud account policy is created locally under ~/.oci/lacework_cloud_account.json. You can post this file to Lacework via the Lacework CLI to create your integration:
-
-```
- lacework api post /api/v2/CloudAccounts -d "$(cat ~/.oci/lacework_cloud_account.json)"
-```
-
-
-| Note          |
+You can run the Terraform scripts locally or via an OCI cloud shell. We recommend using the OCI cloud shell, since it comes with Terraform pre-installed and provides for authentication.
+
+## Prepare to run Terraform locally (Option 1)
+
+Before using the Terraform script, you must create the required environment variables to access your OCI tenant. The recommended way to do this is:
+
+1. Clone this github repo. For example:
+   ```
+   git clone https://github.com/lacework-dev/scripts.git
+   ```
+2. Change directory to the `lw-oci-integration/tf` folder and copy the `vars.tfvars.example` to `terraform.tfvars`.
+3. Update the values in this file to align with your administrative OCI user access.
+    ```
+    region="<YOUR HOME REGION  (e.g. us-ashburn-1)>"
+    tenancy_ocid="<YOUR OCI TEANCY OCID>"
+    user_ocid="<YOUR OCI USER OCID>"
+    fingerprint="<YOUR KEY FINGER PRINT>"
+    private_key_path="<YOUR KEY KEY PATH (e.g. ~/.oci/myocikey.pem)>"
+    ```
+4. Install the Lacework CLI (if it is not already installed) and configure with your API key. For more information, see the [Lacework CLI Guide](https://docs.lacework.net/cli/).
+
+## Prepare to run Terraform via OCI Cloud Shell (Option 2)
+
+1. Open the OCI cloud shell. For example:
+   1. Log in to the OCI console.
+   2. Click on the '<>' icon (developer tools), located on the top right of the OCI web console.
+   3. Select Cloud Shell.
+2. Clone this github repo. For example:
+    ```
+    git clone https://github.com/lacework-dev/scripts.git
+    ```
+3. Change directory to the `lw-oci-integration/tf` folder and copy `vars.tfvars.example` to `terraform.tfvars`.
+4. OCI auth is automatically handled by the cloud shell. To use this, we modify a couple of files:
+   * In `main.tf`, edit the provider section so that it looks like this (setting `tenancy_ocid` and `region` accordingly):
+     
+     ```
+     provider "oci" {
+       tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaatxph3zpu3jkem4cseejktwllb6dgsam7cx3jy63pazejzobvjqqq"
+       region = "us-sanjose-1"
+     } 
+     
+     ```
```

Comment:
Hmmm, actually the `main.tf` today uses the variables from tfvars to set the provider details:
```
provider "oci" {
  tenancy_ocid     = var.tenancy_ocid
  user_ocid        = var.user_ocid
  fingerprint      = var.fingerprint
  private_key_path = var.private_key_path
  region           = var.region
}
```

I don't think we need to or should recommend hard coding these values.

Reply:
Thanks - I see main.tf change has been reverted and instead the instruction for cloudshell is to set non-required variables to null:
```
     user_ocid=null
     fingerprint=null
     private_key_path=null
```